daemon/logger/jsonfilelog/jsonfilelog.go

```diff
@@ -38,36 +42,123 @@ func New(ctx logger.Context) (logger.Logger, error) {
 	if err != nil {
 		return nil, err
 	}
+	var capval int64
+	capacity, ok := ctx.Config["max-size"]
+	if !ok {
```

Comment:
you cant do that. I transform the data in the map to a different representation. The map holds string data like `1k (or) 1G`. I convert these into bytes.